geode-management/src/main/java/org/apache/geode/management/api/ClusterManagementResult.java

```diff
@@ -127,19 +130,11 @@ public String getStatusMessage() {
     return statusMessage;
   }
 
-  /**
-   * Returns the full path (not including http://server:port) by which this result can be referenced
-   * via REST
-   */
-  public String getUri() {
-    return uri;
-  }
-
   /**
    * for internal use only
    */
-  public void setUri(String uri) {
-    this.uri = uri;
+  public void setLinks(Links links) {
```

Comment:
This is a problem. You can't have "internal" classes external APIs.

Reply:
After some review, we concluded that Links just needs to be public, as all Configuration objects are required to implement `abstract Links getLinks()`.  We found low value in separating out an interface, as this is a pretty simple data class.  Even if we make it immutable in a future refactoring, there would still be no need for an interface.